tests/run_tests.py

```
@@ -2959,6 +2959,195 @@ def boom(path):
+    # clean case: two malformed entries are candidate FPs beyond a budget of 1 -> exactly 1 FP, no crash
+    clean = s.score_case({"defects": [], "fp_budget": 1}, ["x", None])
+    assert clean["fp"] == 1 and clean["fp_candidates"] == 2
+
+
+def t_eval_score_case_one_finding_one_defect():
+    # E1-S2 (PR #44 Codex P1 regression): a single reviewer finding must credit at most one defect.
+    # When two must_detect defects sit within tolerance (or share a tag) in the same file, one finding
+    # scored both defects as TP -> tp:2, inflating detection. Assignment is now one-to-one.
+    s = _import_score()
+    D = lambda n, floor="high": {"defect_id": "d%d" % n, "must_detect": True, "severity_floor": floor,
+                                 "locators": [{"file": "a.py", "line_range": [n, n]}], "root_cause_tags": ["t%d" % n]}
```

Ruff (0.16.1)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `D` as a `def`

(E731)